### PR TITLE
libsonata-report: update with new repository

### PR DESCRIPTION
**Pull request by alkino**

### CODE REVIEW

File: var/spack/repos/builtin/packages/libsonatareport/package.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Libsonatareport(CMakePackage):
+    """
+    `libsonatareport` provides C++ API for reading SONATA reports
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/libsonatareport/package.py]:
```suggestion
    `libsonatareport` provides C++ API for writing SONATA reports
```

File: var/spack/repos/builtin/packages/libsonatareport/package.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Libsonatareport(CMakePackage):
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/libsonatareport/package.py]: As discussed on Slack, we should remove old package and make this as libsonata-report

File: var/spack/repos/builtin/packages/libsonatareport/package.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        ]
+        if self.spec.satisfies('+mpi'):
+            result.extend([
+                '-DCMAKE_C_COMPILER:STRING={0}'.format(
+                    self.spec['mpi'].mpicc
+                ),
+                '-DCMAKE_CXX_COMPILER:STRING={0}'.format(
+                    self.spec['mpi'].mpicxx
+                ),
+            ])
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/libsonatareport/package.py]: question / nitpick : Is this required? If we are doing right thing with FindMPI and linking necessary MPI libraries, this shouldn't be needed.

**Review by pramodk — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/libsonatareport/package.py (hunk 3 above)
Comment (alkino) [reply var/spack/repos/builtin/packages/libsonatareport/package.py]: no idea, we only discuss about that months ago

**Comment by alkino** [comment]
Should we delete it?

**Comment by matz-e** [comment]
Yes.

**Pull request closed without merging**